MapEditor/Components/MovementHandler.cs

```diff
@@ -0,0 +1,269 @@
+            Vector2 position,
+            float stopRadius)
+        {
+            var target = destinations.FirstOrDefault();
+            if (HasArrived(position, target, stopRadius))
+            {
+                destinations.RemoveAt(0);
+                destinations.Add(target);
+            }
+            return target;
+        }
+
+        private static ITarget GetRandomTarget(IList<ITarget> destinations,
+            Vector2 position,
+            float stopRadius)
+        {
+            var target = destinations.FirstOrDefault();
+            if (HasArrived(position, target, stopRadius))
+            {
+                //Generate a random target in a small arear
+                destinations.RemoveAt(0);
+                //destinations.Insert(0, )
+            }
+            return target;
+        }
+
+        private static bool HasArrived(Vector2 position, ITarget target, float stopRadius)
+        {
+            return target != null && position.Distance(target.Position) >= stopRadius;
+        }
+
+        private static Vector2 Move(Vector2 targetPosition,
+            Vector2 position,
+            Vector2 velocity,
+            float slowRadius,
+            float maxVelocity,
+            float force,
+            float mass)
+        {
+            //subtract the position from the target to get the vector from the vehicles position to the target. 
+            var distance = targetPosition - position/* / Data.MaxVelocity*/;
+
+            //Normalize it then multiply by max speed to get the maximum velocity from your position to the target.
+            var desiredVelocity = distance.Truncate(maxVelocity); // distance.Normalize()
+
+            //if (distance.Length() <= slowRadius)
+            //desiredVelocity =* (maxVelocity * distance / slowRadius);
+            //else
+            //    desiredVelocity.ScaleBy(movement.MaxVelocity);
+
+            //subtract velocity from the desired velocity to get the force vector
+            var steering = desiredVelocity - velocity;
+
+            //divide the steeringForce by the mass(which makes it the acceleration), 
+            steering = steering.Truncate(force) / mass;
+
+            return steering;
+        }
+
+        private static Vector2 Intercept(Vector2 targetPosition,
+            Vector2 targetVelocity,
+            Vector2 position,
+            Vector2 velocity,
+            float slowRadius,
+            float maxVelocity,
+            float force,
+            float mass)
+        {
+            var distance = targetPosition - position;
+
+            var updatesNeeded = distance / maxVelocity;
+
+            var targetVector = targetVelocity * updatesNeeded;
+
+            return Move(targetVector,
+                position,
+                velocity,
+                slowRadius,
+                maxVelocity,
+                force,
+                mass);
+        }
+
+        private void Follow(ITarget target)
```

Comment:
There is examples online for this, will be important for group movement.
Leader + Seperation